### PR TITLE
CleanCode & Bridge

### PR DESCRIPTION
**Pull request by l0kr**

### CODE REVIEW

File: src/main/java/edu/kis/vh/nursery/containers/IntArrayStack.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package edu.kis.vh.nursery.containers;
+
+// użycie klawisza ctrl oraz kursora myszki pozwala przejść zarówno do deklaracji
+// jak i implementacji
+
+public class IntArrayStack implements IntContainer {
+    static final int capacity = 12;
```
Comment (kowallus) [inline src/main/java/edu/kis/vh/nursery/containers/IntArrayStack.java]: PLEASE FIX: Constants should be written UPPERCASE (CAPACITY)

File: src/main/java/edu/kis/vh/nursery/containers/IntArrayStack.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    int total = -1;
+    int[] NUMBERS = new int[capacity];
```
Comment (kowallus) [inline src/main/java/edu/kis/vh/nursery/containers/IntArrayStack.java]: PLEASE FIX: Array is not a constant (lowercase)

File: src/main/java/edu/kis/vh/nursery/containers/IntArrayStack.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
+package edu.kis.vh.nursery.containers;
+
+// użycie klawisza ctrl oraz kursora myszki pozwala przejść zarówno do deklaracji
+// jak i implementacji
+
+public class IntArrayStack implements IntContainer {
+    static final int capacity = 12;
+    int total = -1;
```
Comment (kowallus) [inline src/main/java/edu/kis/vh/nursery/containers/IntArrayStack.java]: Another constant.

File: src/main/java/edu/kis/vh/nursery/containers/IntArrayStack.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    public IntArrayStack() {
+    }
+
+    public void countIn(int in) {
+        if (!isFull())
+            NUMBERS[++total] = in;
+    }
+
+    public boolean callCheck() {
+        return total == empty;
```
Comment (kowallus) [inline src/main/java/edu/kis/vh/nursery/containers/IntArrayStack.java]: Extracting this constant is done incorrectly. (This is not the same empty as e.g. in peakaboo method)

File: src/main/java/edu/kis/vh/nursery/containers/IntArrayStack.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    public boolean isFull() {
+        return total == 11;
```
Comment (kowallus) [inline src/main/java/edu/kis/vh/nursery/containers/IntArrayStack.java]: PLEASE FIX: Another constant. It is related to CAPACITY (yet, the name is misinforming)

File: src/main/java/edu/kis/vh/nursery/containers/IntContainer.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
+package edu.kis.vh.nursery.containers;
+
+public interface IntContainer {
+
+    static final int empty = 0;
```
Comment (kowallus) [inline src/main/java/edu/kis/vh/nursery/containers/IntContainer.java]: PLEASE FIX: Constants should be written UPPERCASE.

File: src/main/java/edu/kis/vh/nursery/containers/IntLinkedList.java [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+    public void setLast(Node last) {
+        this.last = last;
+
+
+    }
+
+
+    private class Node {
+
+        private int value;
```
Comment (kowallus) [inline src/main/java/edu/kis/vh/nursery/containers/IntLinkedList.java]: PLEASE FIX: Immutable - should be final (setter unnecessary)

File: src/main/java/edu/kis/vh/nursery/CountingOutRhymer.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+public class CountingOutRhymer {
+
+  private IntContainer intContainer;
+
+    public CountingOutRhymer(IntContainer intContainer) {
+        super();
+        this.intContainer = intContainer;
+    }
+
+    public CountingOutRhymer() {
```
Comment (kowallus) [inline src/main/java/edu/kis/vh/nursery/CountingOutRhymer.java]: PLEASE FIX: Remove default constructor and fix places that use it (e.g. tests, factories)

File: src/main/java/edu/kis/vh/nursery/factory/ArrayStackRhymersFactory.java [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+import edu.kis.vh.nursery.HanoiRhymer;
+
+public class ArrayStackRhymersFactory implements RhymersFactory {
+    @Override
+    public CountingOutRhymer getStandardRhymer() {
+        return new CountingOutRhymer();
+    }
+
+    public CountingOutRhymer getFalseRhymer() {
+        return new CountingOutRhymer();
```
Comment (kowallus) [inline src/main/java/edu/kis/vh/nursery/factory/ArrayStackRhymersFactory.java]: PLEASE FIX: getFalseRhymer should use different IntContainer than other factory methods.

**Review by kowallus — CHANGES_REQUESTED** [review]
Required updates are marked using 'PLEASE FIX:' prefix. In case of repeated issues usually only one place is marked, but fixes are expected to be made to similar places.

File: src/main/java/edu/kis/vh/nursery/containers/IntContainer.java (hunk 6 above)
Comment (l0kr) [reply src/main/java/edu/kis/vh/nursery/containers/IntContainer.java]: ✔️

File: src/main/java/edu/kis/vh/nursery/containers/IntArrayStack.java (hunk 5 above)
Comment (l0kr) [reply src/main/java/edu/kis/vh/nursery/containers/IntArrayStack.java]: ✔️

File: src/main/java/edu/kis/vh/nursery/containers/IntArrayStack.java (hunk 1 above)
Comment (l0kr) [reply src/main/java/edu/kis/vh/nursery/containers/IntArrayStack.java]: ✔️

File: src/main/java/edu/kis/vh/nursery/containers/IntArrayStack.java (hunk 2 above)
Comment (l0kr) [reply src/main/java/edu/kis/vh/nursery/containers/IntArrayStack.java]: ✔️

File: src/main/java/edu/kis/vh/nursery/containers/IntArrayStack.java [hunk 10]
```diff
@@ -4,39 +4,40 @@
 // jak i implementacji
 
 public class IntArrayStack implements IntContainer {
-    static final int capacity = 12;
-    int total = -1;
-    int[] NUMBERS = new int[capacity];
+    static final int CAPACITY = 12;
+    int TOTAL = -1;
```
Comment (kowallus) [inline src/main/java/edu/kis/vh/nursery/containers/IntArrayStack.java:8]: This is not a constant!

File: src/main/java/edu/kis/vh/nursery/containers/IntArrayStack.java [hunk 11]
```diff
@@ -4,39 +4,40 @@
...
 
     public boolean callCheck() {
-        return total == empty;
+        return numbers[TOTAL] == null ;
+
     }
 
     public boolean isFull() {
-        return total == 11;
+        return TOTAL == 11;
```
Comment (kowallus) [inline src/main/java/edu/kis/vh/nursery/containers/IntArrayStack.java:25]: NOT FIXED: "PLEASE FIX: Another constant. It is related to CAPACITY (yet, the name is misinforming)"
11 is a constant (not TOTAL)

**Review by kowallus — COMMENTED** [review]
Almost OK

**Pull request closed without merging**